routing/routing_integration_tests/bicycle_route_test.cpp

```diff
@@ -241,87 +330,123 @@ UNIT_TEST(Spain_Madrid_ElevationsDetour)
+  // (651m via a tertiaty + service in this case).
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
-                                   mercator::FromLatLon(51.9960994, 5.67350176), {0.0, 0.0},
-                                   mercator::FromLatLon(51.9996861, 5.67299133), 428.801);
+                                   mercator::FromLatLon(68.0192918, 32.9576269), {0.0, 0.0},
+                                   mercator::FromLatLon(68.0212968, 32.9632167), 323.951 /* expectedRouteMeters */);
+}
 
-  // The difference here (end points are almost equal) because of some barrier=gate without access.
-  // https://www.openstreetmap.org/node/6993853766
+UNIT_TEST(Russia_UseTrunk_AvoidGasStationsDetours)
+{
+  /// @todo(pastk): still OM detours into many petrol stations along the trunk.
+  CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
+                                   mercator::FromLatLon(55.9132468, 39.1453188), {0.0, 0.0},
+                                   mercator::FromLatLon(55.9146268, 39.153307), 549.911 /* expectedRouteMeters */);
+}
 
-  /// @todo Make correct tunnels elevation (linear between start and finish).
+UNIT_TEST(Netherlands_IgnoreCycleBarrier_WithoutAccess)
+{
+  // There is a barrier=cycle_barrier in the beginning of the route
+  // and it doesn't affect routing as there is no explicit bicycle=no.
+  // https://github.com/organicmaps/organicmaps/issues/3920
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
-                                   mercator::FromLatLon(51.3822, -2.3886), {0.0, 0.0},
-                                   mercator::FromLatLon(51.3574666, -2.31526436), 8583.27);
+                                   mercator::FromLatLon(51.9960994, 5.67350176), {0.0, 0.0},
+                                   mercator::FromLatLon(51.9996861, 5.67299133), 428.801);
+}
 
+UNIT_TEST(UK_ForbidGates_WithoutAccess)
+{
+  // A barrier=gate without explicit access leads to a long detour.
+  // https://www.openstreetmap.org/node/6993853766
+  /// @todo OSRM/Valhalla/GraphHopper ignore such gates,
+  /// see https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=51.3579329%2C-2.3137701%3B51.3574666%2C-2.3152644
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
-                                   mercator::FromLatLon(51.3822, -2.3886), {0.0, 0.0},
-                                   mercator::FromLatLon(51.3576973, -2.31416085), 11131.6);
+                                   mercator::FromLatLon(51.3579329, -2.3137701), {0.0, 0.0},
+                                   mercator::FromLatLon(51.3574666, -2.31526436), 1149.28);
 }
 
-// https://github.com/organicmaps/organicmaps/issues/7257
-UNIT_TEST(AvoidConstruction)
+UNIT_TEST(Australia_AvoidConstruction)
 {
   // Will not work when the bridge will be finished.
```

Comment:
maybe having synthetic tests would be better? Rather than specifying location, specify small routing graph/OSM data within test itself?

or specify date and use attic data to fetch OSM data frozen on some specific date? It would prevent tests rotting because someone changed OSM data (at cost of not spotting that preferred tagging changed for something, but this seems much less valuable and should be rare and deprecationists will usually create an issue to complain)

Reply:
There are some tests with embedded map data, though maybe not routing tests.
I guess its just too time-consuming to set them up.
Right, @vng ?

Reply:
>  I guess its just too time-consuming to set them up.

if it is possible than scripting extraction from OSM data should be fairly easy

or maybe even save bunch of .osm files

(StreetComplete has small synthetic tests as when it was using live OSM data tests were getting broken all the time resulting in them being ignored, if someone points me to format/example code I can prepare some for bicycle routing)

Reply:
There are many *.osm samples in `data/test_data/osm/`. Check their usage in the code.

If we'll use such excerpts for routing then how would we visualize them and build routes and reason about?
Right now check the test in the desktop app - its more or less easy to reason which way is better, try to put intermediate points to check alternatives, check output of alternative routers on osm.org...

Maybe synt tests will be suitable for strict yes/no tests only like - a constructed road is a no go, cycleway=opposite should allow going in any direction, etc.

Reply:
yeap sounds good!

Reply:
great! is there any such test already? Can you point me to one?

Reply:
I don't know if there are such routing tests already. @vng?

There are some other tests which seem to setup mock mwms, e.g. `search/search_integration_tests/smoke_test.cpp`